# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webspeech/synth/nsSpeechTask.cpp
# Commit: 1c7a70ea55e8
# Full Hash: 1c7a70ea55e881a42d6bbe2375945f8b532edbb5
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-02-26 09:27:57
# Regressor Bug: 1535617
# File Overlap Count: 1
# Description:
#   Bug 1535617: Part 2 - Clear weak references for most cycle collected objects on unlink. r=mccr8
#   
#   This covers most cycle collected objects which support weak references, but
#   not the ones which inherit from a cycle collected class and don't do any cycle
#   collection on their own.
# ==============================================================================

diff -r 275198c0a966 -r 1c7a70ea55e8 dom/media/webspeech/synth/nsSpeechTask.cpp
--- a/dom/media/webspeech/synth/nsSpeechTask.cpp	Tue Feb 25 19:44:27 2020 +0000
+++ b/dom/media/webspeech/synth/nsSpeechTask.cpp	Tue Feb 25 19:44:39 2020 +0000
@@ -22,7 +22,8 @@
 
 // nsSpeechTask
 
-NS_IMPL_CYCLE_COLLECTION(nsSpeechTask, mSpeechSynthesis, mUtterance, mCallback);
+NS_IMPL_CYCLE_COLLECTION_WEAK(nsSpeechTask, mSpeechSynthesis, mUtterance,
+                              mCallback)
 
 NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(nsSpeechTask)
   NS_INTERFACE_MAP_ENTRY(nsISpeechTask)